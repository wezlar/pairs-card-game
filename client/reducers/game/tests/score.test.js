import actionCreator from '../../../../testHelpers/actionCreator';

import ACTION_TYPES from '../../../actions/constants';
import reducer, { getInitialState } from '../score';

describe(`score.js reducer`, () => {
  const initialState = getInitialState();

  describe('initial state', () => {
    test('it matches the snapshot - default', () => {
      expect(initialState).toMatchSnapshot();
    });
  });

  describe(`the reducer`, () => {
    test('it returns initial state by default', () => {
      expect(reducer(undefined, {})).toEqual(initialState)
    });
    
    test(`it doesn't change state unexpectedly`, () => {
      const fakeAction = actionCreator('NOT_A_REAL_ACTION_TYPE')('test-payload');
      expect(reducer(undefined, fakeAction)).toEqual(initialState)
    });

    test(`status is correct for ${ACTION_TYPES.UPDATE_SCORE}`, () => {
      const payload = { score: 5 };
      let action = actionCreator(ACTION_TYPES.UPDATE_SCORE)(payload)
      expect(reducer(undefined, action)).toEqual(5);
      action = actionCreator(ACTION_TYPES.UPDATE_SCORE)(payload)
      expect(reducer(5, action)).toEqual(10);
    });
  });
});