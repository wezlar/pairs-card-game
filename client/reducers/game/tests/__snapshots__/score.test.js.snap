// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`score.js reducer initial state it matches the snapshot - default 1`] = `0`;
